timesketch/frontend/src/components/Explore/TimelineChip.vue

```diff
@@ -14,28 +14,320 @@ See the License for the specific language governing permissions and
+            <span class="icon is-small">
+            <i class="fas fa-edit"></i>
+            </span>
+            <span>Rename</span>
+          </b-dropdown-item>
+          <b-dropdown-item aria-role="listitem" v-if="timelineStatus === 'ready'" @click="showColorPicker()">
+            <span class="icon is-small">
+            <i class="fas fa-palette"></i>
+            </span>
+            <span>Change color</span>
+          </b-dropdown-item>
+          <b-dropdown-item aria-role="listitem" v-if="timelineStatus === 'ready'" @click="showAnalyzerModal = !showAnalyzerModal">
+            <span class="icon is-small">
+            <i class="fas fa-history"></i>
+            </span>
+            <span>Analyzer logs</span>
+          </b-dropdown-item>
+          <b-dropdown-item aria-role="listitem" @click="remove()" class="is-danger">
+            <span class="icon is-small">
+            <i class="fas fa-trash"></i>
+            </span>
+            <span>Delete</span>
+          </b-dropdown-item>
+        </b-dropdown>
+      </span>
+    </span>
   </div>
 </template>
 
 <script>
+import Vue from 'vue'
+import { Chrome } from 'vue-color'
+import _ from 'lodash'
+
+import TsAnalyzerHistory from '../Analyze/AnalyzerHistory'
+
+import EventBus from '../../main'
+
 export default {
-  props: ['timeline'],
+  components: {
+    'color-picker': Chrome,
+    TsAnalyzerHistory,
+  },
+  props: ['timeline', 'eventsCount', 'isSelected', 'isEmptyState'],
+  data() {
+    return {
+      initialColor: {},
+      newColor: '',
+      newTimelineName: '',
+      timelineStatus: '',
+      colorPickerActive: false,
+      showInfoModal: false,
+      showEditModal: false,
+      showAnalyzerModal: false,
+      isDarkTheme: false,
+    }
+  },
   computed: {
-    timelineColorStyle() {
-      let backgroundColor = this.timeline.color
+    meta() {
+      return this.$store.state.meta
+    },
+    datasourceErrors() {
+      return this.timeline.datasources.filter(datasource => datasource.error_message)
+    },
+    failedImportsMessage() {
+      return this.datasourceErrors.length + " failed imports"
+    },
+  },
+  methods: {
+    showColorPicker() {
+      this.$refs.colorPicker.click()
+    },
+    rename() {
+      this.showEditModal = false
+      this.$emit('save', this.timeline, this.newTimelineName)
+    },
+    remove() {
+      if (confirm('Delete the timeline?')) {
+        this.$emit('remove', this.timeline)
+      }
+    },
+    updateColor: _.debounce(function(color) {
+      this.newColor = color.hex
+      if (this.newColor.startsWith('#')) {
+        this.newColor = this.newColor.substring(1)
+      }
+      Vue.set(this.timeline, 'color', this.newColor)
+      this.$emit('save', this.timeline)
+    }, 300),
+    toggleTheme: function() {
+      this.isDarkTheme = !this.isDarkTheme
+    },
+    timelineColor(timeline) {
```

Comment:
May I suggest renaming this to `getTimelineStyle`? You're returning more than just the color

Reply:
Good catch! Will make the change right away :)